src/transformers/models/bloom/configuration_bloom.py

```diff
@@ -0,0 +1,163 @@
+        attention_softmax_in_fp32 (`bool`, *optional*, defaults to `True`):
+            If set to `True` and the `dtype` is set to `float16` it will scale the input of the Softmax function to
+            `fp32`
+        hidden_dropout (`float`, *optional*, defaults to 0.1):
+            Dropout rate of the dropout function in `bias_dropout_fusion`
+        attention_dropout (`float`, *optional*, defaults to 0.1):
+            Dropout rate applied to the attention probs
+        scale_attn_weights (`bool`, *optional*, defaults to `True`):
+            Scale attention weights by dividing by sqrt(hidden_size)..
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should return the last key/values attentions (not used by all models).
+        scale_attn_by_inverse_layer_idx (`bool`, *optional*, defaults to `False`):
+            Whether to additionally scale attention weights by `1 / layer_idx + 1`.
+        reorder_and_upcast_attn (`bool`, *optional*, defaults to `False`):
+            Whether to scale keys (K) prior to computing attention (dot-product) and upcast attention
+            dot-product/softmax to float() when training with mixed precision.
+        pretraining_tp (`int`, *optional*, defaults to `1`):
+            Tensor parallelism rank used during pretraining with Megatron. Please refer to [this
+            blogpost](https://huggingface.co/docs/transformers/parallelism) to understand more about it. This value is
+            necessary to ensure exact reproducibility of the pretraining results. Please refer to [this
+            issue](https://github.com/pytorch/pytorch/issues/76232)
+
+    Example:
+
+    ```python
+    >>> from transformers import BloomModel, BloomConfig
+
+    >>> # Initializing a Bloom configuration
+    >>> configuration = BloomConfig()
+
+    >>> # Initializing a model from the configuration
+    >>> model = BloomModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "bloom"
+    keys_to_ignore_at_inference = ["past_key_values"]
+    attribute_map = {
+        "num_hidden_layers": "n_layer",
+        "n_head": "num_attention_heads",
+        "hidden_size": "n_embed",
+    }
+
+    def __init__(
+        self,
+        vocab_size=250880,
+        hidden_size=64,
+        n_layer=2,
+        n_head=8,
+        n_inner=None,
+        masked_softmax_fusion=True,
+        layer_norm_epsilon=1e-5,
+        initializer_range=0.02,
+        use_cache=False,
+        bos_token_id=1,
+        eos_token_id=2,
+        apply_residual_connection_post_layernorm=False,
+        bias_dropout_fusion=True,
+        hidden_dropout=0.0,
+        attention_dropout=0.0,
+        attention_softmax_in_fp32=True,
+        pretraining_tp=1,  # TP rank used when training with megatron
+        dtype="bfloat16",
```

Comment:
I really don't think we want this -> cc @sgugger here

Note that:

a) This will become very confusing with the `torch_dtype` of `configuration_utils.py`
b) We don't want the model's dtype to be dependent on a dtype attribute of the config. Maybe we should have a longer discussion on this as we might not have found a good conclusion on Slack: https://huggingface.slack.com/archives/C01N44FJDHT/p1652349718584159

cc @sgugger @stas00

Reply:
100% agree

Reply:
yes, we just had this discussion a few hours ago. and yes, it should be input.dtype based most of the time unless it's a special case. (e.g. dtype of the input to the word embedding is int64).

Reply:
the config will have `torch_dtype` automatically set when the model is saved.

Reply:
Made an attempt of fixing that in a6436894e9fdcf6b4f0d7cc3e52da53158daa6e3 
Now the `dtype` is automatically tied to `torch_dtype`